LICENSE.txt

```diff
@@ -2252,3 +2252,13 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
+
+--------------------------------------------------------------------------------
+
+3rdparty dependency mathworks/libmexclass is redistributed as a dynamically
+linked shared library in certain binary distributions, like the MATLAB
+distribution.
+
+Copyright: 2022-2024 The MathWorks, Inc. All rights reserved.
+Homepage: https://github.com/mathworks/libmexclass
+License: 3-clause BSD
```

Comment:
We don't need this because we don't bundle mathworks/libmexclass (our source archive, apache-arrow-X.Y.Z.tar.gz, doesn't include mathworks/libmexclass.).

Reply:
Thanks for clarifying how to assemble the `NOTICE.txt` and `LICENSE.txt` files, @kinow 

For reference - we based our original changes off the fact that other third party libraries that are bundled and dynamically linked against in certain binary distributions are noted in the `LICENSE.txt`.

For example:

```
3rdparty dependency zlib is redistributed as a dynamically linked shared
library in certain binary distributions, like the python wheels.
...
```

Reply:
@kinow - yes, sorry about that! Thanks!

Reply:
Ah, we should recheck existing entries such as the zlib's one in `LICENSE.txt`...

> Also, we noted the following instructions in https://infra.apache.org/licensing-howto.html#permissive-deps:

It seems that this description is for "bundled" dependency not "non-bundled" dependency.

https://infra.apache.org/licensing-howto.html#step-by-step

> For any [bundled](https://infra.apache.org/licensing-howto.html#bundled-vs-non-bundled) dependency, consider whether LICENSE and/or NOTICE need to be modified. Do not modify LICENSE or NOTICE for non-bundled dependencies.

Reply:
Created #40895 as a follow up to address this.

Reply:
Marking this as resolved.